### PR TITLE
Add viralflye

### PR DESCRIPTION
**Pull request by TomHarrop**
Add viral genome recovery pipeline [viralflye](https://github.com/Dmitry-Antipov/viralFlye)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by TomHarrop** [comment]
@BiocondaBot please add label

**Comment by TomHarrop** [comment]
@BiocondaBot please fetch artifacts

File: recipes/viralflye/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  number: 0
+  noarch: python
+
+requirements:
+  build:
+    - git
+  host:
+    - python ==3.10.8
+  run:
+    - bcftools ==1.16
```
Comment (bgruening) [inline recipes/viralflye/meta.yaml:22]: If possible do not pin dependencies that strict. If you do that those packages can hardly be mixed with others in the same environment.

File: recipes/viralflye/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    - tabix ==1.11
+    - vcflib ==1.0.3
+    - viralverify ==1.1
+    # - viralcomplete NO TAGGED RELEASE
+
+test:
+  commands:
+    # both exit 1
+    - viralcomplete -h &> /dev/null || [[ "$?" == 1 ]]
+    - viralFlye.py  2> /dev/null || [[ "$?" == 1 ]]
```
Comment (bgruening) [inline recipes/viralflye/meta.yaml:44]: can you grep for expected output here?

**Review by bgruening — COMMENTED** [review]
Thanks @TomHarrop. A few comments inline.

File: recipes/viralflye/meta.yaml (hunk 1 above)
Comment (TomHarrop) [reply recipes/viralflye/meta.yaml]: Unpinned in https://github.com/bioconda/bioconda-recipes/pull/39987/commits/e2510f0926c1f08529288b23bef9250575a8171a. I only pinned the deps for a reproducible build, but we have the biocontainer for that :)

File: recipes/viralflye/meta.yaml (hunk 2 above)
Comment (TomHarrop) [reply recipes/viralflye/meta.yaml]: Fixed https://github.com/bioconda/bioconda-recipes/pull/39987/commits/99ab3963c2e1d84274f1d7a96b3cf280fe938f51

File: recipes/viralflye/build.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# install viralcomplete from source
+(
+    git clone \
+        https://github.com/ablab/viralComplete \
+        $SRC_DIR/viralComplete
+    cd $SRC_DIR/viralComplete || exit 1
+    git checkout -f db9ffa7
+    mv bin/viralcomplete $PREFIX/bin/
+    mv blast_db $PREFIX/
```
[the remark this answers is not in the document]
Comment (TomHarrop) [reply recipes/viralflye/build.sh]: Fixed https://github.com/bioconda/bioconda-recipes/pull/39987/commits/df7a36a19858aa4e0feb3d7404b0f62afde819dd / https://github.com/bioconda/bioconda-recipes/pull/39987/commits/d1c0283787d96c381cd7ef0baf74ab7ae5a73218

**Comment by TomHarrop** [comment]
@BiocondaBot please fetch artifacts

**Comment by TomHarrop** [comment]
@BiocondaBot please fetch artifacts

File: recipes/viralflye/build.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomHarrop) [reply recipes/viralflye/build.sh]: Ahh ok, I had to create the data and blast_db directories in build and copy the content in post-link.

**Comment by TomHarrop** [comment]
Thanks @bgruening I think this is OK now.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@TomHarrop sorry for the late reply. Please feel free to ping me on matrix next time.

**Comment by TomHarrop** [comment]
Hi @bioconda/core I don't think this deployed properly. The package is not on anaconda and the container is not on quay.io.

Guessing from the [error message](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=30732&view=results) I think the artifacts had been removed when the PR got merged. Is there a way to re-build and deploy again?

**Pull request merged**